src/components/[guild]/AccessHub/hooks/useEditGuildPlatform.ts

```diff
@@ -13,16 +20,30 @@ const useEditGuildPlatform = ({
   onSuccess?: () => void
 }) => {
   const { id, mutateGuild } = useGuild()
+  const { guildPlatform: originalGuildPlatform } = useGuildPlatform(guildPlatformId)
 
   const showErrorToast = useShowErrorToast()
 
-  const submit = async (signedValidation: SignedValidation) =>
-    fetcher(`/v2/guilds/${id}/guild-platforms/${guildPlatformId}`, {
-      method: "PUT",
-      ...signedValidation,
-    })
+  const fetcherWithSign = useFetcherWithSign()
+  const submit = async (data: Partial<GuildPlatform>) =>
+    fetcherWithSign([
+      `/v2/guilds/${id}/guild-platforms/${guildPlatformId}`,
+      {
+        method: "PUT",
+        /**
+         * We need to send the whole platformGuildData here, since our API replaces
+         * this object with the data we send inside the request body
+         */
+        body: {
+          platformGuildData: {
```

Comment:
We just send the `platformGuildData` to the API here, but I think that's fine, since we won't need to edit anything else on a `guildPlatform`.